### PR TITLE
Single Video and Trending page

### PR DESCRIPTION
**Pull request by surbhi84**
I added single video Page that opens upon clicking a video card, and a trending page that displays the trending videos.

### CODE REVIEW

File: src/components/VideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    <div className="flex flex-row flex-wrap p-8">
+      {videoList.map(
+        ({ id, title, creator, views, uploadedOn, about, img, avatar }) => (
+          <div
+            className="h-96 w-1/4 p-3"
+            key={uuid()}
+            onClick={() => {
+              navigate(`/video/${id}`, {
+                state: {
+                  id,
```
Comment (akshay-rajput) [inline src/components/VideoCard.jsx:19]: put into your global state as a variable "currentlyViewingVideo", also handle a case when page refreshes, you make an api call and store data in currentlyViewingVideo

File: src/components/VideoCard.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { useNavigate } from "react-router-dom";
+import { v4 as uuid } from "uuid";
+import { useVideos } from "hooks";
+
+export function VideoCard() {
+  const navigate = useNavigate();
+  const { videoList } = useVideos();
+
```
Comment (akshay-rajput) [inline src/components/VideoCard.jsx:8]: in useEffect check the id of the video if it has changed, fetch video data

File: src/pages/FullScreen.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { BiLike, BiDislike } from "react-icons/bi";
```
Comment (akshay-rajput) [inline src/pages/FullScreen.jsx:1]: rename file to SingleVideoPage or VideoPage and refactor wherever necessary

File: src/pages/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { VideoCard } from "components";
+
+export function Home() {
+  return <VideoCard />;
```
Comment (akshay-rajput) [inline src/pages/Home.jsx:4]: rename to videoListing and videoCard should be for a single video component

File: src/components/Navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { useMenu } from "hooks";
```
Comment (SJTGSHIVAM) [inline src/components/Navbar.jsx:1]: Nice usage of custom hooks I must say

File: src/pages/Trending.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      <hr className="my-4 color-slate-300" />
+
+      <div className="flex flex-row">
+        {trendingVids.map(
+          ({ id, title, creator, views, uploadedOn, about, img, avatar }) => (
+            <div
+              className="h-2/4 w-2/5 p-3"
+              key={uuid()}
+              onClick={() => {
+                navigate(`/video/${id}`, {
```
Comment (SJTGSHIVAM) [inline src/pages/Trending.jsx:22]: Nice usage of sharing sate while page transition.

**Review by SJTGSHIVAM — APPROVED** [review]
I dont find any major issues, code looks clean. I really appreciate the different approaches you have used to get the job done. It is good to merge. Let me know if we can sync up some time need to take some tips on tailwind as well.

**Comment by surbhi84** [comment]
I have incorporated the all the necessary changes in the next branch.

**Pull request merged**